include/openssl/bio.h

```diff
@@ -452,6 +447,23 @@ OPENSSL_EXPORT int BIO_set_mem_buf(BIO *bio, BUF_MEM *b, int take_ownership);
 OPENSSL_EXPORT int BIO_set_mem_eof_return(BIO *bio, int eof_value);
 
 
+// BIO close flags.
+//
+// These can be used as symbolic arguments when a "close flag" is passed to a
+// BIO function.
+
+// BIO_NOCLOSE will not free the backing buffer on BIO close
+#define BIO_NOCLOSE 0
+
+// BIO_CLOSE (default) will free the backing buffer on BIO close, whereas
+#define BIO_CLOSE 1
+
+// BIO_FP_TEXT will cause the file to be opened as a text file (this is only
+// relevant on Windows due to CRLF endings) instead of the default behavior of
+// opening the file in binary mode.
+#define BIO_FP_TEXT 0x10
```

Comment:
This is a good question. This shouldn't matter if all we're doing is calling `_setmode` after open, as windows `fopen` [considers](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-wfopen?view=msvc-170#generic-text-routine-mappings) "text" mode to be the default and will honor `b` when passed to `fopen`. however, if we're doing more than `_setmode`, I think we need to perform this check-and-set-`BIO_FP_TEXT`.

If the fseek/ftell tests I write indicate that  we need to call `setvbuf`, then I'll implement this. Else, I don't think we need it.

Reply:
The test I added isn't able to reproduce the issue, so I don't think we need the `setvmode` and I'd like to avoid porting over OpenSSL's legacy cruft for dealing with fixed bugs.

To your initial comment, I guess the question is whether we want to pass through to the default behavior in this case (i.e. windows treating `mode` str w/o `'b'` set as "text mode")? I believe that this default behavior will need to remain stable through time as by now, surely many windows applications have come to rely on it. That said, it probably wouldn't _hurt_ to be explicit and set `BIO_FP_TEXT` internally in this case. What do you think?